Index: src/hrda/model/Department.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/hrda/model/Department.java b/src/hrda/model/Department.java
new file mode 100644
--- /dev/null	(date 1753024306641)
+++ b/src/hrda/model/Department.java	(date 1753024306641)
@@ -0,0 +1,35 @@
+package hrda.model;
+
+public class Department {
+
+    private Long id;
+    private String department;
+
+    public Department() {
+    }
+
+    public Department(Long id, String department) {
+        this.id = id;
+        this.department = department;
+    }
+
+    public Department(String departmentName) {
+        this.department = departmentName;
+    }
+
+    public Long getId() {
+        return id;
+    }
+
+    public void setId(Long id) {
+        this.id = id;
+    }
+
+    public String getDepartment() {
+        return department;
+    }
+
+    public void setDepartment(String department) {
+        this.department = department;
+    }
+}
Index: src/hrda/serviceImpls/EmployeeServiceImpl.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/hrda/serviceImpls/EmployeeServiceImpl.java b/src/hrda/serviceImpls/EmployeeServiceImpl.java
new file mode 100644
--- /dev/null	(date 1753026065407)
+++ b/src/hrda/serviceImpls/EmployeeServiceImpl.java	(date 1753026065407)
@@ -0,0 +1,141 @@
+package hrda.serviceImpls;
+
+import hrda.model.Department;
+import hrda.model.Employee;
+import hrda.repository.DBConnection;
+import hrda.service.EmployeeService;
+
+import java.sql.*;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.stream.Stream;
+
+public class EmployeeServiceImpl implements EmployeeService {
+
+    Stream stream;
+
+    private final DBConnection dbConnection;
+
+    private final DepartmentService departmentService;
+
+    public EmployeeServiceImpl(DBConnection dbConnection, DepartmentService departmentService) {
+        this.dbConnection = dbConnection;
+        this.departmentService = departmentService;
+    }
+
+    @Override
+    public List<Employee> getAllEmployee() throws SQLException {
+        List<Employee> employeeList = new ArrayList<>();
+
+        String query = "SELECT * FROM Employee";
+
+        try (Connection connection = dbConnection.getConnection();
+             Statement statement = connection.createStatement();
+             ResultSet resultSet = statement.executeQuery(query)) {
+            while (resultSet.next()) {
+                Long id = resultSet.getLong("id");
+                String employeeName = resultSet.getString("employee_name");
+                Long departmentId = resultSet.getObject("department_id", Long.class);
+                String departmentName = null;
+                if (departmentId != null && departmentService != null) {
+                    departmentName = departmentService.getDepartmentName(departmentId);
+                }
+
+                Double salary = resultSet.getDouble("salary");
+                Integer experienceYears = resultSet.getInt("experience_years");
+
+                Department department = new Department(departmentId, departmentName);
+                Employee employee = new Employee(id, employeeName, department, salary, experienceYears);
+                employeeList.add(employee);
+            }
+        }
+        //No catch clause because I have already handled exception in methods declaration
+
+
+        return employeeList;
+    }
+
+    @Override
+    public List<Employee> getEmployeesBySalary(Double salary) throws SQLException {
+        List<Employee> employeeList = new ArrayList<>();
+        String query = "SELECT * FROM Employee WHERE salary >= 50000";
+        try(
+                Connection connection = dbConnection.getConnection();
+                PreparedStatement preparedStatement = connection.prepareStatement(query);
+                ResultSet resultSet = preparedStatement.executeQuery()){
+            while (resultSet.next()){
+                Long id  = resultSet.getLong("id");
+                String employeeName = resultSet.getString("employee_name");
+                Long departmentId = resultSet.getObject("department_id", Long.class);
+                String departmentName = null;
+                if(departmentId != null && departmentService != null){
+                    departmentName = departmentService.getDepartmentName(departmentId);
+                }
+                Double employeeSalary = resultSet.getDouble("salary");
+                Integer experienceInYears = resultSet.getInt("experience_years");
+                Department department = new Department(departmentId, departmentName);
+                Employee employee = new Employee(id, employeeName, department, salary, experienceInYears);
+                employeeList.add(employee);
+            }
+
+        }
+        //No catch clause because I have already handled exception in methods declaration
+
+        return employeeList;
+    }
+
+    @Override
+    public Double averageSalaryOfDepartment(String departmentName) throws SQLException{
+        Double averageSalary = null;
+        String query = "SELECT AVG(e.salary) AS average_salary " +
+                "FROM Employee e " +
+                "JOIN Department d ON e.department_id = d.id " +
+                "WHERE d.department = ? " +
+                "GROUP BY d.department";
+        try (
+                Connection connection = dbConnection.getConnection();
+                PreparedStatement preparedStatement = connection.prepareStatement(query);
+                ){
+            preparedStatement.setString(1, departmentName);
+            ResultSet resultSet = preparedStatement.executeQuery();
+            while (resultSet.next()){
+                averageSalary = resultSet.getDouble("average_salary");
+            }
+        }
+
+        return averageSalary;
+    }
+
+    @Override
+    public List<Employee> getEmployeesByExpAndSalary(Integer expInYears, Double Salary) {
+        return Collections.emptyList();
+    }
+
+    @Override
+    public boolean addEmployee(String employeeName, String department,Double newSalary,Integer employeeExperience) {
+        String sql = "INSERT INTO Employee (employee_name, department_id, salary, experience_years) VALUES (?, ?, ?, ?)";
+
+        Department departmentByName = departmentService.findDepartmentByName(department);
+        if(departmentByName == null) {
+            departmentService.addDepartment(department);
+            departmentByName = departmentService.findDepartmentByName(department);
+        }
+        try (Connection connection = dbConnection.getConnection();
+             PreparedStatement statement = connection.prepareStatement(sql)) {
+
+            statement.setString(1, employeeName);
+            statement.setObject(2, departmentByName.getId());
+            statement.setDouble(3, newSalary);
+            statement.setInt(4, employeeExperience);
+
+            int rowsInserted = statement.executeUpdate();
+            return rowsInserted > 0;
+
+        } catch (SQLException e) {
+            e.printStackTrace();
+            return false;
+        }
+    }
+
+}
Index: src/hrda/serviceImpls/DepartmentService.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/hrda/serviceImpls/DepartmentService.java b/src/hrda/serviceImpls/DepartmentService.java
new file mode 100644
--- /dev/null	(date 1753026159547)
+++ b/src/hrda/serviceImpls/DepartmentService.java	(date 1753026159547)
@@ -0,0 +1,79 @@
+package hrda.serviceImpls;
+
+import hrda.model.Department;
+import hrda.model.Employee;
+import hrda.repository.DBConnection;
+
+import java.sql.*;
+import java.util.List;
+
+public class DepartmentService {
+
+    Department department = new Department();
+    private final DBConnection dbConnection;
+
+    public DepartmentService(DBConnection dbConnection) {
+        this.dbConnection = dbConnection;
+    }
+
+    public String getDepartmentName(Long departmentId) throws SQLException {
+        String departmentName = null;
+        String query = "SELECT * FROM Department WHERE id = ?";
+        Connection connection = dbConnection.getConnection();
+        PreparedStatement statement = connection.prepareStatement(query);
+        statement.setLong(1, departmentId);
+        ResultSet resultSet = statement.executeQuery();
+        while (resultSet.next()){
+//            Long id = resultSet.getLong("id");
+            departmentName = resultSet.getString("department");
+//            System.out.println("----------------------");
+//            System.out.println("Id: "+id);
+//            System.out.println("Department: "+departmentName);
+        }
+        return departmentName;
+    }
+
+    public Department findDepartmentByName(String departmentName) {
+        String sql = "SELECT id, department FROM Department WHERE department = ?";
+
+        try (Connection connection = dbConnection.getConnection();
+             PreparedStatement statement = connection.prepareStatement(sql)) {
+
+            statement.setString(1, departmentName);
+
+            try (ResultSet rs = statement.executeQuery()) {
+                if (rs.next()) {
+                    Long id = rs.getLong("id");
+                    String name = rs.getString("department");
+                    return new Department(id, name);
+                }
+            }
+
+        } catch (SQLException e) {
+            e.printStackTrace();
+        }
+
+        return null;
+    }
+
+    public boolean addDepartment(String departmentName) {
+        String sql = "INSERT INTO Department (department) VALUES (?)";
+
+        try (Connection connection = dbConnection.getConnection();
+             PreparedStatement statement = connection.prepareStatement(sql)) {
+
+            statement.setString(1, departmentName);
+
+            int rowsInserted = statement.executeUpdate();
+            return rowsInserted > 0;
+
+        } catch (SQLException e) {
+            e.printStackTrace();
+            return false;
+        }
+    }
+
+
+
+
+}
Index: src/hrda/repository/DbOperations.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/hrda/repository/DbOperations.java b/src/hrda/repository/DbOperations.java
new file mode 100644
--- /dev/null	(date 1752943033679)
+++ b/src/hrda/repository/DbOperations.java	(date 1752943033679)
@@ -0,0 +1,4 @@
+package hrda.repository;
+
+public class DbOperations {
+}
Index: src/hrda/service/EmployeeService.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/hrda/service/EmployeeService.java b/src/hrda/service/EmployeeService.java
new file mode 100644
--- /dev/null	(date 1753025107451)
+++ b/src/hrda/service/EmployeeService.java	(date 1753025107451)
@@ -0,0 +1,14 @@
+package hrda.service;
+
+import hrda.model.Employee;
+
+import java.sql.SQLException;
+import java.util.List;
+
+public interface EmployeeService {
+    List<Employee> getAllEmployee() throws SQLException;
+    List<Employee> getEmployeesBySalary(Double salary) throws SQLException;
+    Double averageSalaryOfDepartment(String departmentName) throws SQLException;
+    List<Employee> getEmployeesByExpAndSalary(Integer expInYears, Double Salary);
+    boolean addEmployee(String employeeName, String department,Double newSalary,Integer employeeExperience);
+}
Index: src/hrda/repository/DBConnection.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/hrda/repository/DBConnection.java b/src/hrda/repository/DBConnection.java
new file mode 100644
--- /dev/null	(date 1752985798490)
+++ b/src/hrda/repository/DBConnection.java	(date 1752985798490)
@@ -0,0 +1,41 @@
+package hrda.repository;
+
+import java.sql.Connection;
+import java.sql.DriverManager;
+import java.sql.SQLException;
+
+public class DBConnection {
+    private final String url = "jdbc:mysql://localhost:3306/hrda";
+    private final String user = "atharva.haptiq";
+    private final String password = "pass1234";
+
+    public Connection getConnection() throws SQLException {
+        return DriverManager.getConnection(url, user, password);
+    }
+
+    public static void main(String[] args) {
+        DBConnection db = new DBConnection();
+        try (Connection connection = db.getConnection()) {
+            System.out.println("DB connected successfully!");
+
+            /*
+            String query = "SELECT * FROM Department WHERE id = ?";
+            try (PreparedStatement statement = connection.prepareStatement(query)) {
+                statement.setInt(1, 1);
+                try (ResultSet resultSet = statement.executeQuery()) {
+                    while (resultSet.next()) {
+                        int id = resultSet.getInt("id");
+                        String department = resultSet.getString("department");
+
+                        System.out.println("ID: " + id);
+                        System.out.println("Department: " + department);
+                    }
+                }
+            }
+            */
+
+        } catch (SQLException e) {
+            e.printStackTrace();
+        }
+    }
+}
diff --git a/readme.md b/readme.md
new file mode 100644
