Index: HRDA.iml
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.BaseRevisionTextPatchEP
<+><?xml version=\"1.0\" encoding=\"UTF-8\"?>\n<module type=\"JAVA_MODULE\" version=\"4\">\n  <component name=\"NewModuleRootManager\" inherit-compiler-output=\"true\">\n    <exclude-output />\n    <content url=\"file://$MODULE_DIR$\">\n      <sourceFolder url=\"file://$MODULE_DIR$/src\" isTestSource=\"false\" />\n    </content>\n    <orderEntry type=\"inheritedJdk\" />\n    <orderEntry type=\"sourceFolder\" forTests=\"false\" />\n  </component>\n</module>
===================================================================
diff --git a/HRDA.iml b/HRDA.iml
--- a/HRDA.iml	(revision 40d9843583026e1be3b1936789faae0e28c0ddae)
+++ b/HRDA.iml	(date 1752984133663)
@@ -7,5 +7,14 @@
     </content>
     <orderEntry type="inheritedJdk" />
     <orderEntry type="sourceFolder" forTests="false" />
+    <orderEntry type="module-library" exported="">
+      <library>
+        <CLASSES>
+          <root url="jar://$USER_HOME$/Downloads/mysql-connector/usr/share/java/mysql-connector-j-9.3.0.jar!/" />
+        </CLASSES>
+        <JAVADOC />
+        <SOURCES />
+      </library>
+    </orderEntry>
   </component>
 </module>
\ No newline at end of file
Index: src/Main.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.BaseRevisionTextPatchEP
<+>//TIP To <b>Run</b> code, press <shortcut actionId=\"Run\"/> or\n// click the <icon src=\"AllIcons.Actions.Execute\"/> icon in the gutter.\npublic class Main {\n    public static void main(String[] args) {\n        System.out.println(\"HRDA Project\");\n    }\n}
===================================================================
diff --git a/src/Main.java b/src/Main.java
--- a/src/Main.java	(revision 40d9843583026e1be3b1936789faae0e28c0ddae)
+++ b/src/Main.java	(date 1753026373546)
@@ -1,7 +1,168 @@
-//TIP To <b>Run</b> code, press <shortcut actionId="Run"/> or
-// click the <icon src="AllIcons.Actions.Execute"/> icon in the gutter.
+import hrda.model.Employee;
+import hrda.repository.DBConnection;
+import hrda.service.EmployeeService;
+import hrda.serviceImpls.DepartmentService;
+import hrda.serviceImpls.EmployeeServiceImpl;
+
+import java.sql.SQLException;
+import java.util.List;
+import java.util.Scanner;
+import java.util.stream.Stream;
+
 public class Main {
-    public static void main(String[] args) {
-        System.out.println("HRDA Project");
+    public static void main(String[] args) throws SQLException {
+        Scanner scanner = new Scanner(System.in);
+        int choice;
+
+        Stream stream;
+
+
+        DBConnection dbConnection = new DBConnection();
+        DepartmentService departmentService = new DepartmentService(dbConnection);
+        EmployeeServiceImpl employeeService = new EmployeeServiceImpl(dbConnection,departmentService);
+
+        while (true) {
+            printMenu();
+
+            System.out.print("Enter your choice: ");
+            if (scanner.hasNextInt()) {
+                choice = scanner.nextInt();
+                scanner.nextLine(); // Consume newline
+
+                List<Employee> employeeList;
+                String  departmentName;
+                switch (choice) {
+                    case 1:
+                        System.out.println("Showing all employees...");
+                        employeeList = employeeService.getAllEmployee();
+                        employeeList
+                                .stream()
+                                .forEach(employee -> { // Here the warning I am getting that we can use forEach directly but assessment is on Streams topic so used that
+                                    System.out.println(" ");
+                                    System.out.println("---------------Employee Info------------------");
+                                    System.out.println("Employee Name: "+employee.getEmployeeName());
+                                    System.out.println("Department: "+employee.getDepartment().getDepartment());
+                                    System.out.println("Salary: "+employee.getSalary());
+                                    System.out.println("Experience In Years: "+employee.getExperienceYears());
+                                });
+                        break;
+                    case 2:
+                        System.out.println("Showing employees earning more than ₹50,000...");
+                         employeeList = employeeService.getAllEmployee();
+                         employeeList
+                                 .stream()
+                                 .filter(employee -> employee.getSalary() >= 50000)
+                                 .forEach(employee ->{
+                                     System.out.println(" ");
+                                     System.out.println("---------------Employee Info------------------");
+                                     System.out.println(" ");
+                                     System.out.println("Employee Name: "+employee.getEmployeeName());
+                                     System.out.println("Department: "+employee.getDepartment().getDepartment());
+                                     System.out.println("Salary: "+employee.getSalary());
+                                     System.out.println("Experience In Years: "+employee.getExperienceYears());
+                                 });
+                        break;
+                    case 3:
+                        System.out.println("Grouping employees by department...");
+                        System.out.println("Enter department name: ");
+                        departmentName = scanner.nextLine();
+                        employeeList = employeeService.getAllEmployee();
+                        employeeList
+                                .stream()
+                                .filter(employee -> employee.getDepartment().getDepartment().equals(departmentName))
+                                .forEach(employee -> {
+                                    System.out.println(" ");
+                                    System.out.println("---------------Employee Info------------------");
+                                    System.out.println(" ");
+                                    System.out.println("Employee Name: "+employee.getEmployeeName());
+                                    System.out.println("Department: "+employee.getDepartment().getDepartment());
+                                    System.out.println("Salary: "+employee.getSalary());
+                                    System.out.println("Experience In Years: "+employee.getExperienceYears());
+                                });
+
+                        break;
+                    case 4:
+                        System.out.println("Calculating average salary per department...");
+                        System.out.println("Enter departmentName you want to search for:");
+                        departmentName = scanner.nextLine();
+                        Double averageSalaryOfDepartment = employeeService.averageSalaryOfDepartment(departmentName);
+                        if (averageSalaryOfDepartment != null)System.out.println("Average salary of "+departmentName+" department is: "+averageSalaryOfDepartment);
+                        else System.out.println("No results found for department: " + departmentName);
+                        break;
+                    case 5:
+                        System.out.println("Sorting employees by experience and salary...");
+                        System.out.println("Enter min experience in years you want to sort for:");
+                        Integer experienceInYears = scanner.nextInt();
+                        System.out.println("Enter min salary you want to sort for: ");
+                        Double salary = scanner.nextDouble();
+                        employeeList = employeeService.getAllEmployee();
+                        employeeList
+                                .stream()
+                                .filter(employee -> employee.getExperienceYears()>=experienceInYears &&
+                                        employee.getSalary()>=salary)
+                                .forEach(employee -> {
+                                    System.out.println(" ");
+                                    System.out.println("---------------Employee Info------------------");
+                                    System.out.println(" ");
+                                    System.out.println("Employee Name: "+employee.getEmployeeName());
+                                    System.out.println("Department: "+employee.getDepartment().getDepartment());
+                                    System.out.println("Salary: "+employee.getSalary());
+                                    System.out.println("Experience In Years: "+employee.getExperienceYears());
+                                });
+                        break;
+                    case 6:
+                        System.out.println("Adding employee...");
+                        System.out.println("Enter Name");
+                        String employeeName = scanner.nextLine();
+                        System.out.println("Enter Department Name");
+                        String department = scanner.nextLine();
+                        System.out.println("Enter salary");
+                        Double newSalary = scanner.nextDouble();
+                        System.out.println("Enter experience(in years)");
+                        Integer employeeExperience = scanner.nextInt();
+
+                        if (employeeService.addEmployee(employeeName, department,newSalary,employeeExperience)){
+                            System.out.println("Employee added successfully!");
+                        }
+                        else System.out.println("Employee failed to add");
+                        break;
+                    case 7:
+                        System.out.println("Adding department...");
+                        System.out.println("Add department Name to add");
+                        departmentName = scanner.nextLine();
+                        if(departmentService.addDepartment(departmentName)) System.out.println("Department added succesfully!");
+                        else System.out.println("Department failed to add");
+                        break;
+                    case 8:
+                        System.out.println("Exiting...");
+                        System.exit(0);
+                        break;
+                    default:
+                        System.out.println("Invalid choice. Please try again.");
+                }
+
+            } else {
+                System.out.println("Please enter a valid number.");
+                scanner.nextLine();
+            }
+
+            System.out.println("\nPress Enter to continue...");
+            scanner.nextLine();
+        }
     }
-}
\ No newline at end of file
+
+    private static void printMenu() {
+        System.out.println("------------------------------------------");
+        System.out.println("         H R D A TOOL            ");
+        System.out.println("------------------------------------------");
+        System.out.println("1. Show all employees");
+        System.out.println("2. Show employees earning more than ₹50,000");
+        System.out.println("3. Group employees by department");
+        System.out.println("4. Show average salary per department");
+        System.out.println("5. Sort employees by experience and salary");
+        System.out.println("6. Add an employee");
+        System.out.println("7. Add a department");
+        System.out.println("8. Exit");
+        System.out.println("------------------------------------------");
+    }
+}
Index: src/hrda/model/Employee.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.BaseRevisionTextPatchEP
<+>package hrda.model;\n\npublic class Employee {\n}\n
===================================================================
diff --git a/src/hrda/model/Employee.java b/src/hrda/model/Employee.java
--- a/src/hrda/model/Employee.java	(revision 40d9843583026e1be3b1936789faae0e28c0ddae)
+++ b/src/hrda/model/Employee.java	(date 1753024306633)
@@ -1,4 +1,68 @@
 package hrda.model;
 
 public class Employee {
+    private Long id;
+    private String employeeName;
+    private Department department;
+    private Double salary;
+    private Integer experienceYears;
+
+    public Employee() {
+    }
+
+    public Employee(String employeeName, String departmentName, Double newSalary, Integer employeeExperience) {
+        this.employeeName = employeeName;
+        this.department = new Department(departmentName); // assuming such a constructor exists
+        this.salary = newSalary;
+        this.experienceYears = employeeExperience;
+    }
+
+    public Employee(Long id, String employeeName, Department department, Double salary, Integer experienceYears) {
+        this.id = id;
+        this.employeeName = employeeName;
+        this.department = department;
+        this.salary = salary;
+        this.experienceYears = experienceYears;
+    }
+
+
+    public Long getId() {
+        return id;
+    }
+
+    public void setId(Long id) {
+        this.id = id;
+    }
+
+    public String getEmployeeName() {
+        return employeeName;
+    }
+
+    public void setEmployeeName(String employeeName) {
+        this.employeeName = employeeName;
+    }
+
+    public Department getDepartment() {
+        return department;
+    }
+
+    public void setDepartment(Department department) {
+        this.department = department;
+    }
+
+    public double getSalary() {
+        return salary;
+    }
+
+    public void setSalary(Double salary) {
+        this.salary = salary;
+    }
+
+    public int getExperienceYears() {
+        return experienceYears;
+    }
+
+    public void setExperienceYears(Integer experienceYears) {
+        this.experienceYears = experienceYears;
+    }
 }
